### PR TITLE
Look for ineffective assignments

### PR DESCRIPTION
**Pull request by petemoore**
Has been useful for me, for discovering otherwise hard-to-discover obscure bugs.

### CODE REVIEW

**Comment by petemoore** [comment]
Note, the reported ineffective assignment in travis is fixed in another PR, so let's not fix it in here (otherwise we'll hit a merge conflict later).

**Comment by walac** [comment]
Ok, but this PR needs to rebase to tests pass and merge button be enabled

File: Makefile [hunk 1]
```diff
@@ -32,6 +32,9 @@ check: test
 	# tests should fail if go fmt results in uncommitted code
 	git status --porcelain
 	/bin/bash -c 'test $$(git status --porcelain | wc -l) == 0'
+	go get github.com/gordonklaus/ineffassign
+	"${GOPATH}/bin/ineffassign" .
```
Comment (jonasfj) [inline Makefile]: let's use metalinter instead:
https://github.com/alecthomas/gometalinter

Then we can specify the rules we want to run...

Comment (petemoore) [inline Makefile]: That's being done in #179 but may take a while to land. Let's go with this for now, until #179 lands.

**Comment by jonasfj** [comment]
Hmm, okay... But we really should land #173 soon instead.. it'll create
lots of merge conflicts...

Den 24. feb. 2017 2.51 AM skrev "Pete Moore" <notifications@github.com>:

**Comment by petemoore** [comment]
True! OK, will close. :-)

**Pull request closed without merging**